### ISSUE TITLE
Add High-Level Finetuning API for All Model Types

### ISSUE THREAD

**Issue by ManvithGopu13** [DayInfinity/Vllama#74]
I want a unified finetuning system that works for:

• LLMs
• Vision models
• Audio models
• 3D models
• Classification/regression models

The user should be able to fine-tune with a single command:

• vllama finetune model_path –data dataset.csv

The system should automatically detect:

• Model type
• Required preprocessing
• Proper loss functions
• Proper training configuration
• Proper output formats

finetune should also generate a reproducible training log and export compatible model artifacts.
